cocos/core/renderer/core/render-window.ts

```diff
@@ -256,17 +261,25 @@ export class RenderWindow {
         for (let i = 0; i < this._cameras.length; ++i) {
             if (this._cameras[i] === camera) {
                 this._cameras.splice(i, 1);
+                if (!this._cameras.length) this._detachFromRoot();
                 return;
             }
         }
     }
 
+    private _detachFromRoot () {
+        const windows = legacyCC.director.root.windows;
+        const idx = windows.indexOf(this);
+        if (idx !== -1) windows.splice(idx, 1);
```

Comment:
Modify windows outside of root is not a good idea. It is better modify it in Root.

Reply:
have modified